packages/@cdklabs/cdk-cicd-wrapper/src/stacks/VPCStack.ts

```diff
@@ -76,41 +120,38 @@ export class VPCStack extends cdk.Stack {
    * @returns The created VPC.
    */
   private launchVPCIsolated(props: VPCStackProps) {
+    this.subnetType = ec2.SubnetType.PRIVATE_ISOLATED;
     const vpc = new ec2.Vpc(this, 'vpc', {
       ipAddresses: ec2.IpAddresses.cidr(props.vpcConfig.vpc?.cidrBlock!),
-      restrictDefaultSecurityGroup: true,
+      restrictDefaultSecurityGroup: props.restrictDefaultSecurityGroup || true,
       subnetConfiguration: [
         {
           cidrMask: props.vpcConfig.vpc?.subnetCidrMask,
           name: 'private-isolated',
-          subnetType: ec2.SubnetType.PRIVATE_ISOLATED,
+          subnetType: this.subnetType,
         },
       ],
       maxAzs: props.vpcConfig.vpc?.maxAzs,
     });
 
-    const securityGroup = new ec2.SecurityGroup(this, 'VpcSecurityGroup', {
+    this.securityGroup = new ec2.SecurityGroup(this, 'VpcSecurityGroup', {
       vpc: vpc,
       description: 'Allow traffic between CodeBuildStep and AWS Service VPC Endpoints',
       securityGroupName: 'Security Group for AWS Service VPC Endpoints',
-      allowAllOutbound: true,
+      allowAllOutbound: props.allowAllOutbound || true,
     });
-    securityGroup.addIngressRule(ec2.Peer.ipv4(vpc.vpcCidrBlock), ec2.Port.tcp(443), 'HTTPS Traffic');
 
-    [
-      ec2.InterfaceVpcEndpointAwsService.SSM,
-      ec2.InterfaceVpcEndpointAwsService.STS,
-      ec2.InterfaceVpcEndpointAwsService.CLOUDWATCH_LOGS,
-      ec2.InterfaceVpcEndpointAwsService.CLOUDFORMATION,
-      ec2.InterfaceVpcEndpointAwsService.SECRETS_MANAGER,
-      ec2.InterfaceVpcEndpointAwsService.KMS,
-    ].forEach((service: ec2.InterfaceVpcEndpointAwsService) => {
-      vpc.addInterfaceEndpoint(`VpcEndpoint${service.shortName}`, {
-        service,
-        open: false,
-        securityGroups: [securityGroup],
-      });
-    });
+    this.securityGroup.addIngressRule(ec2.Peer.ipv4(vpc.vpcCidrBlock), ec2.Port.tcp(443), 'HTTPS Traffic');
+
+    props.codeBuildVPCInterfaces && props.codeBuildVPCInterfaces.length > 0
+      ? [...this.defaultCodeBuildVPCInterfaces, ...props.codeBuildVPCInterfaces]
```

Comment:
I would prefer if we would do the merge earlier in the constructor to a parameter, like this.codeBuildVPCInterfaces, and avoid this complexity here.